File: LeetCode/java/1086_High_Five.java
import java.util.*;

class Solution {
    public int[][] highFive(int[][] items) {
        Map<Integer, List<Integer>> map = new HashMap<>();

        for (int[] item : items) {
            if (!map.containsKey(item[0])) {
                map.put(item[0], new ArrayList<>());
            }

            map.get(item[0]).add(item[1]);
        }

        int[][] answer = new int[map.size()][2];
        int idx = 0;

        for (Map.Entry<Integer, List<Integer>> m : map.entrySet()) {
            List<Integer> list = m.getValue();

            Collections.sort(list, Collections.reverseOrder());

            int sum = 0;
            for (int i = 0; i < 5; i++) {
                sum += list.get(i);
            }

            int avg = sum / 5;

            answer[idx][0] = m.getKey();
            answer[idx][1] = avg;
            idx++;
        }

        Arrays.sort(answer, (x, y) -> {
            if (x[0] < y[0]) {
                return -1;
            }

            if (x[0] > y[0]) {
                return 1;
            }

            return 0;
        });

        return answer;
    }
}

File: LeetCode/cpp/1086_High_Five.cpp
class Solution {
public:
	vector<vector<int>> highFive(vector<vector<int>>& items) {
		vector<vector<int>> answer;
		unordered_map<int, vector<int>> um;

		for (vector<int> item : items) {
			um[item[0]].push_back(item[1]);
		}

		for (auto m : um) {
			vector<int> v = m.second;

			sort(v.begin(), v.end(), greater<>());

			int sum = 0;
			for (int i = 0; i < 5; i++) {
				sum += v[i];
			}

			int avg = sum / 5;

			answer.push_back({ m.first, avg });
		}

		sort(answer.begin(), answer.end());

		return answer;
	}
};